ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -81,6 +105,74 @@ const ApproveStaticSimulation = () => {
   );
 };
 
+const SpendingCap = () => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { receivedToken } = useReceivedToken();
+
+  const { tokenAmount, formattedTokenNum, value, pending } =
+    useApproveTokenSimulation(transactionMeta);
+
+  if (pending) {
+    return <Container isLoading />;
+  }
+
+  if (!receivedToken) {
+    return null;
+  }
+
+  return (
+    <ConfirmInfoSection>
+      <ConfirmInfoRow label={t('accountBalance')}>
+        <ConfirmInfoRowText
+          text={`${receivedToken.string} ${receivedToken.symbol}`}
+        />
+      </ConfirmInfoRow>
+
+      {value ? (
+        <>
+          <ConfirmInfoRowDivider />
+
+          <ConfirmInfoRow
+            label={t('spendingCap')}
+            tooltip={t('spendingCapTooltipDesc')}
+          >
+            {tokenAmount === t('unlimited') ? (
+              <Tooltip title={formattedTokenNum}>
+                <ConfirmInfoRowText
+                  text={`${tokenAmount} ${receivedToken.symbol}`}
+                  onEditClick={
+                    transactionMeta.type ===
+                    TransactionType.tokenMethodIncreaseAllowance
+                      ? () => console.log('TODO on a following ticket')
+                      : undefined
+                  }
+                  editIconClassName="edit-spending-cap"
+                />
+              </Tooltip>
+            ) : (
+              <ConfirmInfoRowText
```

Comment:
Is this `ConfirmInfoRowText` also mostly the same as the other branch?

Could we have a a conditional `text` property and conditionally wrap it in the `Tooltip` to avoid the duplication?